app/src/main/java/com/ably/tracking/demo/publisher/ui/main/MainScreen.kt

```diff
@@ -26,11 +29,41 @@ import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
+import com.ably.tracking.demo.publisher.PublisherService
 import com.ably.tracking.demo.publisher.R
+import com.ably.tracking.demo.publisher.common.doOnCreateLifecycleEvent
 import com.ably.tracking.demo.publisher.ui.widget.TextButton
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.rememberPermissionState
 
+import org.koin.androidx.compose.getViewModel
+
+@ExperimentalPermissionsApi
 @Composable
-fun MainScreen(viewModel: MainViewModel) {
+fun MainScreen(
+    activity: ComponentActivity,
+    viewModel: MainViewModel = getViewModel()
+) {
+
+    val locationPermissionState = rememberPermissionState(
+        permission = Manifest.permission.ACCESS_FINE_LOCATION,
+        onPermissionResult = { granted ->
+            onLocationPermissionResult(granted, activity, viewModel)
+        }
+    )
+
+    doOnCreateLifecycleEvent {
+        locationPermissionState.launchPermissionRequest()
+    }
+
+    doOnCreateLifecycleEvent {
+        ContextCompat.startForegroundService(
+            activity,
+            Intent(activity, PublisherService::class.java)
+        )
+    }
```

Comment:
Why 2 separate listeners for the same lifecycle event? :thinking:

Reply:
I think I wanted to highlight that they are two totally unrelated actions 🤔 will change